csrc/host_ir/executor.cpp

```diff
@@ -42,12 +42,27 @@ std::vector<at::Tensor> HostIrExecutor::runWithInput(
   return outputs;
 }
 
+void HostIrExecutor::handle(Stream* stream) {
+  if (streams_.find(stream) == streams_.end()) {
+    auto i = (communicator_ != nullptr && communicator_->is_available())
+        ? communicator_->deviceId()
+        : 0;
+    streams_.insert(
+        {stream,
+         c10::cuda::getStreamFromPool(
+             /* high priority */ true, static_cast<c10::DeviceIndex>(i))});
+  }
+  setCurrentCUDAStream(streams_.at(stream));
```

Comment:
Will this change the stream used for the following communication? Apparently, the communicator constructed outside HostIrExecutor will keep using the same ProcessGroup (and therefore the same underlying stream) regardless of this call.

Reply:
Let me clarify.

In general, ProcessGroups deal two streams:
- an internal stream, not accessible by the user, used to execute nccl kernel
- the user stream, as the one defined through `setCurrentCUDAStream`. The `c10d::Work::wait()`primitive basically synchronizes the user stream and the internal stream.

There is no way to change the internal stream, the only "workaround" to do multi-stream with PGs is to instantiate several PGs. This will be the object of a future PR, in which we will most probably introduce a new IR indicating we need to change ProcessGroup

Reply:
Thanks -- internal streams are what I was asking about.